src/Lie_algebra/Lie_algebra_interface.jl

```diff
@@ -42,23 +42,228 @@ function LieAlgebra(G::LieGroup{𝔽,O}) where {𝔽,O<:AbstractGroupOperation}
+    𝔤::LieAlgebra, c, B::ManifoldsBase.AbstractBasis, ::Nothing
+)
+    G = 𝔤.manifold
+    return get_vector(G.manifold, identity_element(G), c, B)
+end
+@inline function ManifoldsBase._get_vector(
+    𝔤::LieAlgebra, c, B::ManifoldsBase.AbstractBasis, T::Type
+)
     G = 𝔤.manifold
-    get_vector!(base_manifold(G), X, identity_element(G), c, B)
+    return get_vector(G.manifold, identity_element(G, T), c, B)
+end
+
+@doc "$(_doc_get_vector)"
+function ManifoldsBase.get_vector!(
+    𝔤::LieAlgebra, X, c, B::ManifoldsBase.AbstractBasis=DefaultLieAlgebraOrthogonalBasis()
+)
+    return ManifoldsBase._get_vector!(𝔤, X, c, B)
+end
+
+function ManifoldsBase._get_vector!(
+    𝔤::LieAlgebra, X::T, c, B::DefaultLieAlgebraOrthogonalBasis
+) where {T}
+    return get_vector_lie!(𝔤, X, c, number_system(B))
+end
+function ManifoldsBase._get_vector!(
+    𝔤::LieAlgebra, X::T, c, B::ManifoldsBase.AbstractBasis
+) where {T}
+    G = 𝔤.manifold
+    return ManifoldsBase.get_vector!(G.manifold, X, identity_element(G, T), c, B)
+end
+
+@inline function get_vector_lie(𝔤::LieAlgebra, c, N)
+    X = zero_vector(𝔤)
+    return get_vector_lie!(𝔤, X, c, N)
+end
+@inline function get_vector_lie(𝔤::LieAlgebra, c, N, T::Type)
+    X = zero_vector(𝔤, T)
+    return get_vector_lie!(𝔤, X, c, N)
+end
+@inline function get_vector_lie!(𝔤::LieAlgebra, X::T, c, N) where {T}
+    G = 𝔤.manifold
+    return get_vector!(
+        base_manifold(G),
+        X,
+        identity_element(G, T),
+        c,
+        ManifoldsBase.DefaultOrthogonalBasis(N),
+    )
+end
+
+_doc_hat = """
+    hat(G::LieAlgebra, c)
+    hat(G::LieAlgebra, c, T::Type)
+    hat!(G::LieAlgebra, X::T, c)
+
+Compute the hat map ``(⋅)^̂ : $(_tex(:Cal, "V")) → 𝔤`` that maps a vector of coordinates ``$(_tex(:vec, "c")) ∈ $(_tex(:Cal, "V"))``,
+to a tangent vector ``X ∈ $(_math(:𝔤))``.
+The coefficients are given with respect to a specific basis to a tangent vector in the Lie algebra
+
+```math
+X = $(_tex(:sum))_{i∈$(_tex(:Cal,"I"))} c_iB_i,
+```
+
+where ``$(_tex(:Set, "B_i"))_{i∈$(_tex(:Cal,"I"))}`` is a basis of the Lie algebra
+and ``$(_tex(:Cal,"I"))`` a corresponding index set, which is usually ``$(_tex(:Cal,"I"))=$(_tex(:Set,raw"1,\ldots,n"))``.
+Then ``$(_tex(:Cal, "V")) = ℝ^n``.
+
+For the allocating variant, you can specify the type `T` of the tangent vector to obtain,
+in case there are different representations. The first signature produces the default representation.
+
+The computation can be performed in-place of `X`. The inverse of `hat` is [`vee`](@ref).
+Technically, `hat` is a specific case of [`get_vector`](@ref) and is implemented using the
+[`DefaultLieAlgebraOrthogonalBasis`](@ref).
+"""
+
+# function hat end
+@doc "$(_doc_hat)"
+function hat(𝔤::LieAlgebra{𝔽}, c) where {𝔽}
+    return get_vector(𝔤, c, DefaultLieAlgebraOrthogonalBasis(𝔽))
+end
+function hat(𝔤::LieAlgebra{𝔽}, c, T::Type) where {𝔽}
+    return get_vector(𝔤, c, DefaultLieAlgebraOrthogonalBasis(𝔽); tangent_vector_type=T)
+end
```

Comment:
Why do you try using non-real-coefficient bases here? I'd prefer real-coefficient basis to be the default.

Reply:
So basically replace F with R? Fine for me, until now I was just going for the more generic case, since that was what was done before.

Reply:
Yes. I think the cases where someone might want complex coefficients are rare enough that they should not be the default.

Reply:
But that is not the default here, it would just take the coeffs of the Lie group. 
However, we can surely change the default to always be real coefficients if you prefer; then the basis would be(however) be larger in dimension (the one of the complex Lie group)?

For now this is really just the old default I think, but sure we can change that to the more common case.

Reply:
We should never have the coeffs of the manifold dictate the coefficients of the (default) basis of tangent spaces. The basis would be larger but the group having complex coefficients doesn't mean it's a complex manifold (which is a fairly strict requirement; for example the complex circle is not a complex manifold and does not have a complex coefficient basis).

Reply:
Sure, then let's change the default here.